### PR TITLE
feat(tiler-sharp): extract regions before rescaling them when overzooming

### PR DESCRIPTION
**Pull request by blacha**
- test: add tests for very zoomed in tests
- refactor: remove unused tiles
- feat(tiler-sharp): extract regions before rescaling them when overzooming

### CODE REVIEW

File: packages/sprites/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@basemaps/sprites",
-  "version": "6.28.0",
+  "version": "6.27.0",
```
Comment (Wentao-Kuang) [inline packages/sprites/package.json:3]: Why do we down version here?

Comment (blacha) [reply packages/sprites/package.json:3]: ahh must have been a bad rebase.

**Review by Wentao-Kuang — DISMISSED** [review]

File: packages/server/package.json [hunk 2]
```diff
@@ -47,7 +47,7 @@
     "@basemaps/landing": "^6.28.1"
   },
   "dependencies": {
-    "sharp": "^0.30.2"
+    "sharp": "^0.30.6"
```
Comment (Wentao-Kuang) [inline packages/server/package.json:50]: No "@types/sharp" required here?

Comment (blacha) [reply packages/server/package.json:50]: its not actually used here, but its used as a way of forcing the dependency to be installed.

We bundle the code a single js file. which removes all the deps from it

**Review by Wentao-Kuang — APPROVED** [review]

**Pull request merged**